lib/ansible/modules/cloud/vmware/vmware_content_deploy_template.py

```diff
@@ -0,0 +1,195 @@
+        self.template_service = self.api_client.vcenter.vm_template.LibraryItems
+        self.template_name = self.params.get('template')
+        self.vm_name = self.params.get('name')
+        self.datacenter = self.params.get('datacenter')
+        self.folder = self.params.get('folder')
+        self.resourcepool = self.params.get('resource_pool')
+        self.datastore = self.params.get('datastore')
+        self.pyv = PyVmomi(module=module)
+
+    def deploy_vm_from_template(self):
+        # Find the datacenter by the given datacenter name
+        datacenter_id = self.pyv.find_datacenter_by_name(datacenter_name=self.datacenter)
+        if not datacenter_id:
+            self.module.fail_json(msg="Failed to find the datacenter %s" % self.datacenter)
+        self.datacenter_id = datacenter_id._moId
+        # Find the datastore by the given datastore name
+        self.folder_id = self.get_folder_by_name(self.content_service, self.datacenter, self.folder)
+        if not self.folder_id:
+            self.module.fail_json(msg="Failed to find the folder %s" % self.folder)
+        # Find the resourcepool by the given resourcepool name
+        self.resourcepool_id = self.get_resource_pool_by_name(self.content_service, self.datacenter, self.resourcepool)
+        if not self.resourcepool_id:
+            self.module.fail_json(msg="Failed to find the resource_pool %s" % self.resourcepool)
+        # Find the datastore by the given datastore name
+        datastore_id = self.pyv.find_datastore_by_name(datastore_name=self.datastore)
+        if not datastore_id:
+            self.module.fail_json(msg="Failed to find the datastore %s" % self.datastore)
+        self.datastore_id = datastore_id._moId
+        # Find the LibraryItem by the given LibraryItem name
+        self.library_item_id = self.get_library_item_by_name(self.content_service, self.template_name)
+        if not self.library_item_id:
+            self.module.fail_json(msg="Failed to find the library Item %s" % self.template_name)
+        # Create VM placement specs
+        self.placement_spec = LibraryItems.DeployPlacementSpec(folder=self.folder_id,
+                                                               resource_pool=self.resourcepool_id)
+        self.vm_home_storage_spec = LibraryItems.DeploySpecVmHomeStorage(datastore=to_native(self.datastore_id))
+        self.disk_storage_spec = LibraryItems.DeploySpecDiskStorage(datastore=to_native(self.datastore_id))
+        self.deploy_spec = LibraryItems.DeploySpec(name=self.vm_name,
+                                                   placement=self.placement_spec,
+                                                   vm_home_storage=self.vm_home_storage_spec,
+                                                   disk_storage=self.disk_storage_spec
+                                                   )
+        vm_id = self.template_service.deploy(self.library_item_id, self.deploy_spec)
```

Comment:
Please check if VM with the given name should not exist in the given folder or datacenter before proceeding.